stages/test/test_systemd_unit_create.py

```diff
@@ -100,24 +113,55 @@ def test_systemd_unit_create(tmp_path, stage_module, unit_type, unit_path, expec
             }
         }
     }
+
+    # create units in the tree for systemd-analyze verify
+    unit_section = options["config"]["Unit"]
```

Comment:
What do you think about tweaking this to explicitly create the required systemd files? It is a bit more work to keep it up to date but it is also less magic and more readable (IMHO):
```python
diff --git a/stages/test/test_systemd_unit_create.py b/stages/test/test_systemd_unit_create.py
index 41763d1f..cc49d14c 100644
--- a/stages/test/test_systemd_unit_create.py
+++ b/stages/test/test_systemd_unit_create.py
@@ -115,22 +115,17 @@ def test_systemd_unit_create(tmp_path, stage_module, unit_type, unit_path, expec
     }
 
     # create units in the tree for systemd-analyze verify
-    unit_section = options["config"]["Unit"]
-    target_path = tmp_path / "usr/lib/systemd/system"
-    os.makedirs(target_path, exist_ok=True)
-    for unit_name in unit_section.get("Wants", []) + unit_section.get("Requires", []) + unit_section.get("After", []):
-        with open(target_path / unit_name, mode="w", encoding="utf-8") as fp:
-            # if it's an empty file it get detected as "masked" and fails verification
-            fp.write("[Unit]\n")
-
-    # create units in the tree for systemd-analyze verify
-    binaries = ["mkdir", "touch"]
-    bin_path = tmp_path / "usr/bin"
-    os.makedirs(bin_path, exist_ok=True)
-    for bin_name in binaries:
-        with open(bin_path / bin_name, mode="w", encoding="utf-8") as fp:
-            fp.write("")
-        os.chmod(bin_path / bin_name, mode=0o755)
+    systemd_dir = "usr/lib/systemd/system/"
+    testutil.make_fake_tree(tmp_path, {
+        systemd_dir + "basic.target": "[Unit]\n",
+        systemd_dir + "sysinit.target": "[Unit]\n",
+        systemd_dir + "paths.target": "[Unit]\n",
+        systemd_dir + "sockets.target": "[Unit]\n",
+        "usr/bin/mkdir": "fake-mkdir",
+        "usr/bin/touch": "fake-touch",
+    })
+    for bin_name in ["mkdir", "touch"]:
+        os.chmod(tmp_path / "usr/bin" / bin_name, mode=0o755)
 
     expected_unit_path = tmp_path / expected_prefix / "create-directory.service"
     expected_unit_path.parent.mkdir(parents=True, exist_ok=True)

```

Reply:
I did do it like this at first then thought I'd be smart but clarity is better than fake smartness, so sure.

Reply:
Done.